addOns/automation/src/main/java/org/zaproxy/addon/automation/ExtensionAutomation.java

```diff
@@ -242,6 +243,22 @@ public void registerAutomationJob(AutomationJob job) {
     public void unregisterAutomationJob(AutomationJob job) {
         this.jobs.remove(job.getType());
         this.sortedJobs.remove(job);
+
+        Set<Integer> keys = plans.keySet();
+        for (Integer key : keys) {
+            System.out.println(key);
+            AutomationPlan plan = plans.get(key);
+            System.out.println(plan.getJobIndex(job));
+            System.out.println(job);
+
+            List<AutomationJob> jobs = plan.getJobs();
```

Comment:
Could use an advanced for and avoid the intermediate storage

Reply:
Actually I did try to use the advanced for loop. But it was giving me an exception called `ConcurrentModificationException`, which was telling me to not change the iterator object (`planJob`). Although, I wasn't changing it but maybe the function which I was passing that on was changing it. So the only solution I could find on the forums was to just switch to the old for loops.

Reply:
Does an advanced for work with the intermediate store? Ex: leave this line assigning to “jobs” then iterate it instead of “plan.getJobs”.

Reply:
> Does an advanced for work with the intermediate store?
I think, no. It is still giving me that exception on run-time.

Is using the traditional for loop a bad practice though?

Reply:
Nope, that’s fine. Just trying to simplify 😉